File: server/models/project.js
import mongoose from "mongoose";

const projectSchema=new mongoose.Schema({
    name:String,
    description:String,
    createdBy:{
        type:mongoose.Types.ObjectId,
        ref:"User"
    },
    createdAt:Date,
    deadline:Date,
    tasks:[{
        type:mongoose.Types.ObjectId,
        ref:"Task"
    }]
})

const Project=mongoose.model("projects",projectSchema)

export default Project;

File: client/team_track/lib/models/project.dart

import 'package:team_track/models/task.dart';
import 'package:team_track/models/user.dart';

class Project {
  late String name;
  late String description;
  late User createdBy;
  late DateTime createdAt;
  late DateTime deadline;
  late List<Task> tasks;

  Project({
    required this.name,
    required this.description,
    required this.createdBy,
    required this.createdAt,
    required this.deadline,
    required this.tasks,
  });
}